### PR TITLE
Refactored coloring of statuses and percents (Contingent on standardizing Obz statuses)

### PR DESCRIPTION
**Pull request by RobertAKARobin**
- A single "#color_of" helper method that takes a variety of inputs
- Colors are rendered with `style` tags on HSL scale instead of with CSS classes
- Specs included
- We won't talk about how much math this took the first time around

### CODE REVIEW

File: spec/helpers/application_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        @sub.update!(status: "missing")
+        expect(color_of(@sub, @options)).to eq(0)
+      end
+      it "returns yellow for incomplete" do
+        @sub.update!(status: "incomplete")
+        expect(color_of(@sub, @options)).to eq(60)
+      end
+      it "returns green for complete" do
+        @sub.update!(status: "complete")
+        expect(color_of(@sub, @options)).to eq(120)
```
Comment (andrewsunglaekim) [inline spec/helpers/application_helper_spec.rb]: where are these numbers coming from the equals expectations of `0, 60, 120`

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      end
+      it "returns red for 80%" do
+        expect(color_of(80, @options)).to eq(0)
+      end
+      it "returns red for 70%" do
+        expect(color_of(70, @options)).to eq(0)
+      end
+    end
+  end
+end
```
Comment (andrewsunglaekim) [inline spec/helpers/application_helper_spec.rb]: is there a way to test for an irrational average obseration?

File: spec/helpers/application_helper_spec.rb (hunk 1 above)
Comment (RobertAKARobin) [inline spec/helpers/application_helper_spec.rb]: They're HSL hue colors: `0` is red, `120` is green.

File: spec/helpers/application_helper_spec.rb (hunk 2 above)
Comment (RobertAKARobin) [inline spec/helpers/application_helper_spec.rb]: You sure you mean irrational numbers? That's like `√-i`

Comment (andrewsunglaekim) [inline spec/helpers/application_helper_spec.rb]: i guess what im asking is say we have 3 observations 2 green 1 red that will return something like 4/3 how does that translate into a hue and can we test for that?

**Comment by RobertAKARobin** [comment]
@andrewsunglaekim Tests added! https://github.com/ga-dc/garnet/pull/539/files#diff-31c50b16a4c9f23be71c49d30afcf7edR75

**Comment by andrewsunglaekim** [comment]
yeah, this looks great. I want to merge it in, but honestly, when i looked at it myself, and then with jesse, i still wasn't exactly sure what was going on. I have checked out the branch and see that it works though! :+1: from me but would love @adambray || @mattscilipoti  to pair on this #tooadvanced...yet?

**Comment by jshawl** [comment]
:+1:

**Pull request merged**